src/metabase/automagic_dashboards/core.clj

```diff
@@ -1047,21 +1057,30 @@
                                           :table_id   table-id}))))
        (get-in question [:dataset_query :query :aggregation])))
 
-(defn- collect-breakout-fields
+(s/defn ^:private collect-breakout-fields :- (s/maybe [(type Field)])
   [root question]
-  (map (comp (partial ->field root)
-             first
-             filters/collect-field-references)
-       (get-in question [:dataset_query :query :breakout])))
+  (for [breakout     (get-in question [:dataset_query :query :breakout])
+        field-clause (take 1 (filters/collect-field-references breakout))
+        :let         [field (->field root field-clause)]
+        :when        field]
+    field))
 
 (defn- decompose-question
   [root question opts]
-  (map #(automagic-analysis % (assoc opts
-                                :source       (:source root)
-                                :query-filter (:query-filter root)
-                                :database     (:database root)))
-       (concat (collect-metrics root question)
-               (collect-breakout-fields root question))))
+  (letfn [(analyze [x]
+            (try
+              (automagic-analysis x (assoc opts
+                                           :source       (:source root)
+                                           :query-filter (:query-filter root)
+                                           :database     (:database root)))
+              (catch Throwable e
+                (throw (ex-info (tru "Error decomposing question: {0}" (ex-message e))
+                                {:root root, :question question, :object x}
+                                e)))))]
+    (into []
+          (comp cat (map analyze))
+          [(collect-metrics root question)
+           (collect-breakout-fields root question)])))
```

Comment:
Rewrote this a bit to make it easier to read and add a try-catch that rethrows with extra info if it fails to make debugging easier.